### PR TITLE
Support converting P-Code IRSBs to AIL blocks

### PR DESCRIPTION
**Pull request by mborgerson**
This is a work-in-progress P-Code IRSB to AIL block converter, with some light refactoring to support both VEX and P-Code in harmony. I have a couple things to fix before this should be merged (e.g. calls, stack pointer analysis stuff, and possibly supporting some more instructions), but in my testing this has worked well for small functions.

<details>
<summary>
A very trivial example
</summary>

Decomp using P-Code (with some angr changes improving block optimization)

```c
void block_A()
{
    unsigned int ir_1;
    unsigned int ir_2;

    if (33 == ECX<4>)
    {
        ir_1 = 0;
    }
    else
    {
        ir_2 = 1;
    }
    return;
}
```

Decomp using VEX

```c
void block_A()
{
    BOT ir_0;
    unsigned int ir_1;
    unsigned int ir_2;

    if (33 != ir_0)
    {
        ir_1 = 1;
    }
    else
    {
        ir_2 = 0;
    }
    return;
}
```
</details>

~~Needs test cases before merge. Working on it. Pending angr PR merge.~~ A test case identical to the one present for VEX has been added.

### CODE REVIEW

File: ailment/converter_pcode.py [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+            addr = Const(self._manager.next_atom(),
+                None, var_data.offset, self._manager.arch.bits)
+            return Store(self._statement_idx,
+                    addr, value, var_data.size,
+                    self._manager.arch.memory_endness,
+                    ins_addr=self._manager.ins_addr)
+        else:
+            raise NotImplementedError()
+
+
```
Comment (ltfish) [inline ailment/converter_pcode.py:315]: Remove the extra blank line.

**Review by ltfish — CHANGES_REQUESTED** [review]

**Comment by ltfish** [comment]
LGTM! Feel free to merge after you remove that blank line.

File: ailment/converter_pcode.py (hunk 1 above)
Comment (mborgerson) [reply ailment/converter_pcode.py:315]: Will do

Comment (mborgerson) [reply ailment/converter_pcode.py]: Done!

**Review by ltfish — APPROVED** [review]

**Comment by ltfish** [comment]
Awesome. Thanks!

**Pull request merged**